packages/svelte2tsx/src/htmlxtojsx_v2/nodes/InlineComponent.ts

```diff
@@ -0,0 +1,239 @@
+import MagicString from 'magic-string';
+import { BaseNode } from '../../interfaces';
+import { surroundWithIgnoreComments } from '../../utils/ignore';
+import { transform, TransformationArray } from '../utils/node-utils';
+
+/**
+ * Handles Svelte components as well as svelte:self and svelte:component
+ *
+ * Children of this element should call the methods on this class to add themselves to the correct
+ * position within the transformation.
+ *
+ * The transformation result does not have anything to do with HTMLx, it instead uses plan JS,
+ * leveraging scoped blocks (`{ ... }`). Each element is transformed to something that is
+ * contained in such a block. This ensures we can declare variables inside that do not leak
+ * to the outside while preserving TypeScript's control flow.
+ *
+ * A transformation reads for example like this:
+ * ```
+ * // before
+ * <Comp prop={foo} />
+ * // after
+ * { const $$_Comp = new Comp({ target: __sveltets_2_any(), props: {"prop": foo,}}); }
+ * ```
+ */
+export class InlineComponent {
```

Comment:
This is the core of transformations related components, including svelte:self and svelte:component. I hope the comments make it clear how it works.